### ISSUE TITLE
RuntimeWarning: invalid value encountered in true_divide

### ISSUE THREAD

**Issue by iamlvm** [motokimura/PyTorch_Gaussian_YOLOv3#40]
I got an error while training,

`Gaussian YOLOv3
Gaussian YOLOv3
Gaussian YOLOv3
loading darknet weights.... weights/darknet53.conv.74
using cuda
using tfboard
loading annotations into memory...
Done (t=16.01s)
creating index...
index created!
loading annotations into memory...
Done (t=0.61s)
creating index...
index created!
evaluating...
/home/computer/Desktop/swimmingpool/lib/python3.6/site-packages/numpy/core/fromnumeric.py:83: RuntimeWarning: invalid value encountered in reduce
  return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
/home/computer/Desktop/PyTorch_Gaussian_YOLOv3/utils/utils.py:43: RuntimeWarning: invalid value encountered in subtract
  iou = area / (bbox_area[i] + bbox_area[selec] - area)
/home/computer/Desktop/PyTorch_Gaussian_YOLOv3/utils/utils.py:44: RuntimeWarning: invalid value encountered in greater_equal
  if (iou >= thresh).any():
/home/computer/Desktop/PyTorch_Gaussian_YOLOv3/utils/utils.py:41: RuntimeWarning: invalid value encountered in multiply
  area = np.prod(br - tl, axis=1) * (tl < br).all(axis=1)
/home/computer/Desktop/PyTorch_Gaussian_YOLOv3/utils/utils.py:43: RuntimeWarning: invalid value encountered in true_divide
  iou = area / (bbox_area[i] + bbox_area[selec] - area)`

**Comment by wooys1012** [comment]
I got same error while training
anyone who solve this problem?

**Comment by ZJUzxf** [comment]
The same problem, how to solve this?

**Comment by motokimura** [comment]
They are warnings. 
The code is running without any problems.

As the evaluation runs at the first iteration, you need to wait until it completes.
https://github.com/motokimura/PyTorch_Gaussian_YOLOv3/blob/3b679c8d6fec51f6f689a33d72e6dd85c4195f62/train.py#L160-L172

After the evaluation, training starts.
You should see some training info printed on the terminal every 10 iterations.
https://github.com/motokimura/PyTorch_Gaussian_YOLOv3/blob/3b679c8d6fec51f6f689a33d72e6dd85c4195f62/train.py#L196-L202

**Comment by motokimura** [comment]
Like this:

```
docker@245eaed5efdc:/work$ python train.py --cfg config/gaussian_yolov3_default.cfg --weights_path weights/darknet53.c
onv.74 --tfboard_dir ./log
Setting Arguments.. :  Namespace(cfg='config/gaussian_yolov3_default.cfg', checkpoint=None, checkpoint_dir='checkpoint
s', checkpoint_interval=1000, debug=False, eval_interval=4000, n_cpu=0, tfboard_dir='./log', use_cuda=True, weights_pa
th='weights/darknet53.conv.74')
train.py:57: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. P
lease read https://msg.pyyaml.org/load for full details.
  cfg = yaml.load(f)
successfully loaded config file:  {'MODEL': {'TYPE': 'YOLOv3', 'BACKBONE': 'darknet53', 'ANCHORS': [[10, 13], [16, 30$
, [33, 23], [30, 61], [62, 45], [59, 119], [116, 90], [156, 198], [373, 326]], 'ANCH_MASK': [[6, 7, 8], [3, 4, 5], [0$
 1, 2]], 'N_CLASSES': 80, 'GAUSSIAN': True}, 'TRAIN': {'LR': 0.001, 'MOMENTUM': 0.9, 'DECAY': 0.0005, 'BURN_IN': 1000$
 'MAXITER': 500000, 'STEPS': '(400000, 450000)', 'BATCHSIZE': 4, 'SUBDIVISION': 16, 'IMGSIZE': 608, 'LOSSTYPE': 'l2',
'IGNORETHRE': 0.7, 'GRADIENT_CLIP': 2000.0}, 'AUGMENTATION': {'RANDRESIZE': True, 'JITTER': 0.3, 'RANDOM_PLACING': Tr$
e, 'HUE': 0.1, 'SATURATION': 1.5, 'EXPOSURE': 1.5, 'LRFLIP': True, 'RANDOM_DISTORT': True}, 'TEST': {'CONFTHRE': 0.8,
'NMSTHRE': 0.45, 'IMGSIZE': 416}, 'NUM_GPUS': 1}
effective_batch_size = batch_size * iter_size = 4 * 16
Gaussian YOLOv3
Gaussian YOLOv3
Gaussian YOLOv3
loading darknet weights.... weights/darknet53.conv.74
using cuda
using tfboard
loading annotations into memory...
Done (t=16.17s)
creating index...
index created!
loading annotations into memory...
Done (t=0.55s)
creating index...
index created!
evaluating...
/home/docker/.pyenv/versions/3.6.8/lib/python3.6/site-packages/torch/nn/modules/upsampling.py:129: UserWarning: nn.Up$
ample is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.{} is deprecated. Use nn.functional.interpolate instead.".format(self.name))
/home/docker/.pyenv/versions/3.6.8/lib/python3.6/site-packages/numpy/core/fromnumeric.py:83: RuntimeWarning: invalid $
alue encountered in reduce
  return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
/work/utils/utils.py:43: RuntimeWarning: invalid value encountered in subtract
  iou = area / (bbox_area[i] + bbox_area[selec] - area)
/work/utils/utils.py:44: RuntimeWarning: invalid value encountered in greater_equal
  if (iou >= thresh).any():
/work/utils/utils.py:41: RuntimeWarning: invalid value encountered in multiply
  area = np.prod(br - tl, axis=1) * (tl < br).all(axis=1)
/work/utils/utils.py:43: RuntimeWarning: invalid value encountered in true_divide
  iou = area / (bbox_area[i] + bbox_area[selec] - area)
Loading and preparing results...
DONE (t=52.64s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=78.36s).
Accumulating evaluation results...
DONE (t=27.45s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.028
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.062
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.071
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.066
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.073
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.078
train.py:188: UserWarning: torch.nn.utils.clip_grad_norm is now deprecated in favor of torch.nn.utils.clip_grad_norm_.
  torch.nn.utils.clip_grad_norm(model.parameters(), gradient_clip)
[Iter 0/500000] [lr 0.000000] [Losses: xy 23.074833, wh 73.647552, conf 57359.277344, cls 1206.707275, total 58662.707031, imgsize 608]
[Iter 10/500000] [lr 0.000000] [Losses: xy 94.169800, wh 497.894043, conf 40758.933594, cls 4795.920410, total 46146.917969, imgsize 512]
[Iter 20/500000] [lr 0.000000] [Losses: xy 39.553558, wh 132.618011, conf 40698.324219, cls 2113.736816, total 42984.234375, imgsize 512]
[Iter 30/500000] [lr 0.000000] [Losses: xy 45.515442, wh 203.793671, conf 15874.009766, cls 2281.507568, total 18404.826172, imgsize 320]
[Iter 40/500000] [lr 0.000000] [Losses: xy 13.657714, wh 56.650623, conf 31133.054688, cls 659.041016, total 31862.404297, imgsize 448]
```
